browser/src/layer/tile/CanvasSectionContainer.ts

```diff
@@ -2028,16 +2052,36 @@ class CanvasSectionContainer {
 		this.context.translate(section.myTopLeft[0], section.myTopLeft[1]);
 	}
 
-	private drawSections (frameCount: number = null, elapsedTime: number = null) {
+	private shouldDrawSection (section: CanvasSectionObject) {
+	    return section.isLocated && section.showSection && (!section.documentObject || section.isVisible);
+	}
+
+	private drawSections (frameCount: number = null, elapsedTime: number = null, tileSubset: Set<any> = null) {
 		this.context.setTransform(1, 0, 0, 1, 0, 0);
 
+		var subsetBounds: cool.Bounds = null;
+		// if there is a tileSubset we only want to draw the miniumum region of its bounds
+		if (tileSubset) {
+			var tileSection = <TilesSection>(this.getSectionWithName(L.CSections.Tiles.name));
+			if (tileSection && this.shouldDrawSection(tileSection)) {
+				subsetBounds = tileSection.getSubsetBounds(this.context, tileSubset);
+			}
+			if (subsetBounds) {
+				this.context.save();
```

Comment:
My recollection is (but it is fuzzy) - that it paid dividends performance-wise not to do save/restores - and to avoid clipping wherever we could, and to not translate the canvas - but do the arithmetic ourselves. Worth checking if that is still so - hopefully not.

Reply:
The translate is easily removable if needs be. If it was always just a solid background we could possibly not need a clip, but for calc we have that grid which gets drawn before we draw the cells so I'm not sure we can live without any clipping.